### PR TITLE
Time picker component

### PR DESCRIPTION
**Pull request by louislingatong**

### CODE REVIEW

**Review by sidra2022 — CHANGES_REQUESTED** [review]

File: packages/stencil/src/components/datacom-time-picker/usage.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## Datacom Time Piker
+
+```html
+<form>
+  <div>
+    <DatacomTimePicker
```
[the remark this answers is not in the document]
Comment (louislingatong) [reply packages/stencil/src/components/datacom-time-picker/usage.md]: Done

File: packages/stencil/src/global/global.css [hunk 2]
```diff
@@ -48,16 +48,16 @@
...
-  --dc-date-picker-error-color: var(--dc-token-support-extended-01);
+  --dc-date-time-picker-text-icon-color: var(--dc-token-text-05);
+  --dc-date-time-picker-interactive-icon-color: var(--dc-token-interactive-04);
+  --dc-date-time-picker-ui-color: var(--dc-token-ui-01);
+  --dc-date-time-picker-interactive-hover-color: var(--dc-token-interactive-hover-01);
+  --dc-date-time-picker-primary-text-color: var(--dc-token-text-01);
+  --dc-date-time-picker-secondary-text-color: var(--dc-token-text-05);
+  --dc-date-time-picker-primary-interactive-color: var(--dc-token-interactive-01);
+  --dc-date-time-picker-secondary-interactive-color: var(--dc-token-interactive-02);
+  --dc-date-time-picker-disabled-color: var(--dc-token-disabled-02);
```
[the remark this answers is not in the document]
Comment (louislingatong) [reply packages/stencil/src/global/global.css]: Done

File: packages/stencil/src/global/global.css [hunk 3]
```diff
@@ -48,16 +48,16 @@
...
-  --dc-date-picker-secondary-interactive-color: var(--dc-token-interactive-02);
-  --dc-date-picker-disabled-color: var(--dc-token-disabled-02);
-  --dc-date-picker-error-color: var(--dc-token-support-extended-01);
+  --dc-date-time-picker-text-icon-color: var(--dc-token-text-05);
+  --dc-date-time-picker-interactive-icon-color: var(--dc-token-interactive-04);
+  --dc-date-time-picker-ui-color: var(--dc-token-ui-01);
+  --dc-date-time-picker-interactive-hover-color: var(--dc-token-interactive-hover-01);
+  --dc-date-time-picker-primary-text-color: var(--dc-token-text-01);
+  --dc-date-time-picker-secondary-text-color: var(--dc-token-text-05);
+  --dc-date-time-picker-primary-interactive-color: var(--dc-token-interactive-01);
```
[the remark this answers is not in the document]
Comment (louislingatong) [reply packages/stencil/src/global/global.css]: Done

File: packages/stencil/src/global/global.css [hunk 4]
```diff
@@ -48,16 +48,16 @@
...
-  --dc-date-picker-interactive-icon-color: var(--dc-token-interactive-04);
-  --dc-date-picker-ui-color: var(--dc-token-ui-01);
-  --dc-date-picker-interactive-hover-color: var(--dc-token-interactive-hover-01);
-  --dc-date-picker-primary-text-color: var(--dc-token-text-01);
-  --dc-date-picker-secondary-text-color: var(--dc-token-text-05);
-  --dc-date-picker-primary-interactive-color: var(--dc-token-interactive-01);
-  --dc-date-picker-secondary-interactive-color: var(--dc-token-interactive-02);
-  --dc-date-picker-disabled-color: var(--dc-token-disabled-02);
-  --dc-date-picker-error-color: var(--dc-token-support-extended-01);
+  --dc-date-time-picker-text-icon-color: var(--dc-token-text-05);
```
[the remark this answers is not in the document]
Comment (louislingatong) [reply packages/stencil/src/global/global.css]: Done

File: packages/stencil/src/components/datacom-time-picker/datacom-time-picker.tsx [hunk 5]
```diff
@@ -0,0 +1,1067 @@
...
+      newTime.minute === undefined &&
+      newTime.period === undefined
+    ) {
+      value = this.timeInputElement.value;
+    } else if (
+      newTime.hour !== undefined &&
+      newTime.minute === undefined &&
+      newTime.period === undefined
+    ) {
+      // 00: - 23:
```
[the remark this answers is not in the document]
Comment (louislingatong) [reply packages/stencil/src/components/datacom-time-picker/datacom-time-picker.tsx]: Enhance inline comment for better understanding.

I believe using a block comment "/** */" is good for function/method documentation while inline comment "// " is good for inline code logic.

**Review by sidra2022 — APPROVED** [review]

**Pull request merged**